cli/args/mod.rs

```diff
@@ -819,16 +820,26 @@ impl CliOptions {
 
     if let Some(lockfile) = self.maybe_lockfile() {
       if !lockfile.lock().overwrite {
-        return Ok(Some(
-          snapshot_from_lockfile(lockfile.clone(), api)
```

Comment:
I think the code block below and the one in the language server are essentially the same thing? Could we maintain this snapshot_from_lockfile function in the CLI and then just have that call the deno_npm functions?

Reply:
Sounds good, how about this [9aa1e2d](https://github.com/denoland/deno/pull/20024/commits/9aa1e2ddf9bd378e9a058085fc294b64875bb35b)